### PR TITLE
Fix setQuitOnLastWindowClosed called from instance

### PR DESCRIPTION
**Pull request by EmmanuelMess**
A small fix.

### CODE REVIEW

**Comment by jonaski** [comment]
Thanks. This seem correct. I didn't even know this function was static because it's not listed as static in the documentation.
But maybe change this to: `QApplication::setQuitOnLastWindowClosed`, that's what we do for the other QApplication functions.

**Comment by EmmanuelMess** [comment]
> Thanks. This seem correct. I didn't even know this function was static because it's not listed as static in the documentation.
> But maybe change this to: `QApplication::setQuitOnLastWindowClosed`, that's what we do for the other QApplication functions.

Thanks! Please approve the workflow.

**Comment by EmmanuelMess** [comment]
Fixed a few more functions in main.

File: src/main.cpp [hunk 1]
```diff
@@ -192,7 +192,7 @@ int main(int argc, char *argv[]) {
     }
     return 0;
   }
-  a.setQuitOnLastWindowClosed(false);
+  SingleApplication::setQuitOnLastWindowClosed(false);
```
Comment (jonaski) [inline src/main.cpp:195]: QGuiApplication

File: src/main.cpp [hunk 2]
```diff
@@ -272,7 +272,7 @@ int main(int argc, char *argv[]) {
 #endif
   translations->LoadTranslation("strawberry", ":/translations", language);
   translations->LoadTranslation("strawberry", TRANSLATIONS_DIR, language);
-  translations->LoadTranslation("strawberry", a.applicationDirPath(), language);
+  translations->LoadTranslation("strawberry", SingleApplication::applicationDirPath(), language);
```
Comment (jonaski) [inline src/main.cpp:275]: QCoreApplication

File: src/main.cpp [hunk 3]
```diff
@@ -313,7 +313,7 @@ int main(int argc, char *argv[]) {
 #endif
   QObject::connect(&a, &SingleApplication::receivedMessage, &w, QOverload<quint32, const QByteArray&>::of(&MainWindow::CommandlineOptionsReceived));
 
-  int ret = a.exec();
+  int ret = SingleApplication::exec();
```
Comment (jonaski) [inline src/main.cpp:316]: QCoreApplication

**Comment by jonaski** [comment]
The following can be changed `QCoreApplication::applicationDirPath` (consistent with other places in the code):
https://github.com/strawberrymusicplayer/strawberry/blob/master/ext/libstrawberry-common/core/workerpool.h#L236
If you grep qApp, you'll se a few more instances.
`qApp->exit()` and `qApp->quit()` can use static `QCoreApplication::exit()` and `QCoreApplication::quit()`

**Review by jonaski — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 1 above)
Comment (EmmanuelMess) [reply src/main.cpp:195]: This is buggy, it doesn't work if I change to `QGuiApplication`.

**Comment by EmmanuelMess** [comment]
Fixed!

**Review by jonaski — APPROVED** [review]

**Pull request merged**